packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -644,7 +651,18 @@ abstract class CustomArrayNodeBase<const T extends ImplicitAllowedTypes>
 		getSequenceField(this).removeAt(index);
 	}
 	public removeRange(start?: number, end?: number): void {
```

Comment:
due to our apparent requirement to allow pojo mode arrays to be deep equal to arrays, length has to be an own value property, not a getter, so it has to be provided by the proxy with special logic so it a bit of a mess to deal with and I'd rather not tangle this PR up in that.

Also doing that would be a perf loss as it would have to look up the field twice, which is currently somewhat expensive. Its cheaper to keep to the pattern of lookup the underlying field once in every method.

Reply:
Makes sense - thanks.